pandas/io/parsers.py

```diff
@@ -3606,7 +3606,12 @@ def __next__(self):
                 self.buffer = None
                 line = next(self.f)
```

Comment:
I think you should just be able to use ``.readline()`` always here (and above); file iterators have this property.

Reply:
updated in the next commit.